### PR TITLE
Add Decimals Library

### PR DESCRIPTION
**Pull request by hayesgm**
This patch adds a Decimal library to add arbitrary precision decimals to our Elm code to combat the evils of Rounding Error. We convert each place we use a float into a decimal and try to still have Elm deal with real values instead of "wei" values (which is nice). We currently send flaots back to JavaScript, but we could convert this to wei (as a string) the way out so that there's no rounding errors on either side (since the BN.js library likes getting big strings). Note: the JavaScript interop is weird because we try to use real decimals in Elm (e.g. 5.1 OMG) but the JavaScript library only works on big integers (so we can't represent 5.1 except as 5.1E18 as an integer). Anyway, long-story-short it all works and solves many rounding errors, including the weird usage of -1.

### CODE REVIEW

**Comment by hayesgm** [comment]
FYI, JavaScript passes decimals in as mantissa and exponent in the form of:

```javascript
{
  man: "51222221212212",
  exp: 18
}
```

It's not ideal, but it works pretty well as a way to keep infinite precision.

**Comment by whenceforth** [comment]
need to resolve conflicts

**Comment by hayesgm** [comment]
conflict resolved.

File: src/js/math.js [hunk 1]
```diff
@@ -1,37 +1,10 @@
...
   }
-
-  try {
-    return bn.toNumber()
-  } catch (_err) {
-    return fallback; // too large
-  }
-}
-
-function bigNumberToRealScaled(bn, scale) {
```
Comment (whenceforth) [inline src/js/math.js]: happy to see this function going away!

**Pull request merged**